### PR TITLE
Altering Forced Dependency - Breaking Change on Custom Calendar

### PR DESCRIPTION
**Pull request by callum-mcdata**
### What Does This PR Do

**SEE SUMMARY A FEW COMMENTS DOWN**

In the `metric.sql` macro, there was a forced dependency connecting any use of the metric macro to the default calendar model created by this package.

This is hard-coded into the macro even when it isn't technically correct. Users can override this relationship and provide their own.

As such, I propose removing it until we can figure out how to dynamically reference the correct calendar model.

EDIT: I figured out how to dynamically reference it. See below.

### CODE REVIEW

**Comment by callum-mcdata** [comment]
### Summary

Well it turns out that the reason that the `bigquery` connection was failing is something that @joellabes already identified and included in his `custom_calendar` model. So 🍳 on my face for not looking at the model.


**BREAKING CHANGE**: The `dbt_metrics_calendar_model` will now accept the model name and NOT the `ref`. I know this breaks with the pattern expressed throughout other parts of dbt but this is the only way of consistently pointing the forced dependency in `metric.sql` to point at the right calendar model. 
  - Explanation: In `metric.sql` there is the following line `-- depends on: {{ ref('dbt_metrics_default_calendar') }}`which establishes a DAG relationship to the default calendar even if you're using a custom calendar. 
  - The reason that we have to change the variable to non-ref is because of the way that it gets parsed in the forced dependency. I tried thinking of other ways to represent the forced dependency without changing the format but wasn't able to. 
    - I tried changing the default value to a ref but that fails on the following compilation error `dbt was unable to infer all dependencies for the model "metric_transformation". This typically happens when ref() is placed within a conditional block.`
  - This is a breaking change so we'll want to call it out for anyone who uses it. Given the commented out nature of the forced dependency, we're unable to write in conditional logic for if people use the older version.

If anyone feels that this shouldn't be merged, feel free to let me know here and we can work on other workarounds!

**Comment by jasnonaz** [comment]
This is awesome work, thanks for tracking this down Callum. It's definitely sub-optimal to have a different naming pattern for the custom calendar than basically everything else we do, but it's probably less suboptimal than having errors in the DAG. If Drew feels good about this I'm fine to merge it in.

As far as breaking changes go - @dbeatty10 should be able to walk you through our SemVer strategy and how we can handle cutting a new release for this.

**Comment by callum-mcdata** [comment]
Agreed. I'm not a fan of breaking our pattern but I figure its the worst of two bad options that we can roll-out without needing to overhaul how core handles forced dependencies.

**Comment by callum-mcdata** [comment]
> It's definitely sub-optimal to have a different naming pattern for the custom calendar than basically everything else we do, but it's probably less suboptimal than having errors in the DAG. If Drew feels good about this I'm fine to merge it in.

Agreed. I'm not a fan of breaking our pattern but I figure its the worst of two bad options that we can roll-out without needing to overhaul how core handles forced dependencies.

> As far as breaking changes go - @dbeatty10 should be able to walk you through our SemVer strategy and how we can handle cutting a new release for this.

Sweet - I'll sync with him when he's back next week and I've also figured out if we can remove the get metric calendar macro.

File: macros/get_metric_sql.sql [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 {#-/* TODO: This refs[0][0] stuff is totally ick */#}
 {%- set model = metrics.get_metric_relation(metric.refs[0] if execute else "") %}
-{%- set calendar_tbl = metrics.get_metric_calendar(var('dbt_metrics_calendar_model', "ref('dbt_metrics_default_calendar')")) %}
+{%- set calendar_tbl = ref(var('dbt_metrics_calendar_model', "dbt_metrics_default_calendar")) %}
```
[the remark this answers is not in the document]
Comment (callum-mcdata) [reply macros/get_metric_sql.sql:25]: That is a good find! I did a filesearch on the entire package and it looks like that is the only place where that macro was used. Just pushing a commit that removes it now.

[the remark this answers is not in the document]
Comment (callum-mcdata) [reply macros/get_metric_sql.sql:25]: Removed and passed tests!

**Review by jasnonaz — APPROVED** [review]
🥸

**Pull request merged**